benchmarks/shelley3pools/topology-generator.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+: <<'END_COMMENT'
+Example 3-node topology file:
+
+{
+  "Producers": [
+    {
+      "addr": "127.0.0.1",
+      "port": 3000,
+      "valency": 1
+    },
+    {
+      "addr": "127.0.0.1",
+      "port": 3001,
+      "valency": 1
+    },
+    {
+      "addr": "127.0.0.1",
+      "port": 3003,
+      "valency": 1
+    }
+  ]
+}
+END_COMMENT
+
+. ./configuration/parameters
+
+NNODES=${NNODES:-3}
+
+for current_port in $(seq 3000 $(( 3000 + NNODES - 1 )))
```

Comment:
loop through the node ids: [0 ... NNODES-1]
then, calculate the port: `$((PORTBASE + N))`